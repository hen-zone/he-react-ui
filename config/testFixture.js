import createTestContext from 'react-cosmos-test/enzyme';

export default function testFixture(fixture, variant) {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component;
  mount();
  test(`<${displayName ||
    name} /> rendered correctly with ${variant} fixture`, () =>
    expect(getWrapper()).toMatchSnapshot());
}
