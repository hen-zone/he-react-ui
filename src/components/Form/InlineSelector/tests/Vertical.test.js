import fixture from '../fixtures/Vertical.fixture.js'
import testFixture from '../../../../../config/testFixture'

testFixture(fixture, 'Vertical')
