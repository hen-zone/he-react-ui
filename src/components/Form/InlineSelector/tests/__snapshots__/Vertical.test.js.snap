// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineSelector /> rendered correctly with Vertical fixture 1`] = `
<InlineSelector
  id="demo"
  name="demo"
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "General Exam and Clean",
        "value": "1",
      },
      Object {
        "label": "Toothache",
        "value": "2",
      },
      Object {
        "label": "Loose Filling",
        "value": "3",
      },
      Object {
        "label": "Loose Crown",
        "value": "4",
      },
      Object {
        "label": "Teeth Whitening",
        "value": "5",
      },
    ]
  }
  value={null}
  vertical={true}
>
  <div
    className="outer vertical"
    id="demo"
    name="demo"
  >
    <div
      className="options"
    >
      <div
        className="option"
        key="0"
        onClick={[Function]}
      >
        General Exam and Clean
      </div>
      <div
        className="option"
        key="1"
        onClick={[Function]}
      >
        Toothache
      </div>
      <div
        className="option"
        key="2"
        onClick={[Function]}
      >
        Loose Filling
      </div>
      <div
        className="option"
        key="3"
        onClick={[Function]}
      >
        Loose Crown
      </div>
      <div
        className="option"
        key="4"
        onClick={[Function]}
      >
        Teeth Whitening
      </div>
    </div>
  </div>
</InlineSelector>
`;
